App/SignInWindow.xaml.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Coder.Desktop.App.ViewModels;
+using CommunityToolkit.Mvvm.Input;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Controls.Primitives;
+using Microsoft.UI.Xaml.Data;
+using Microsoft.UI.Xaml.Input;
+using Microsoft.UI.Xaml.Media;
+using Microsoft.UI.Xaml.Navigation;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.Graphics;
+
+namespace Coder.Desktop.App;
+
+/// <summary>
+/// The dialog window to allow the user to sign into their Coder server.
+/// </summary>
+public sealed partial class SignInWindow : Window
+{
+    private const double WIDTH = 600.0;
+    private const double HEIGHT = 300.0;
+
+    private readonly SignInViewModel viewModel;
+    public SignInWindow(SignInViewModel vm)
+    {
+        viewModel = vm;
+        InitializeComponent();
+        var urlPage = new SignInURLPage(this, viewModel);
```

Comment:
Could you just instantiate both of these pages once and store them as private properties on the Window?